docs/cookbook/pii_tutorial.md

```diff
@@ -0,0 +1,260 @@
+
+- **Duplicate Efforts on Problem Resolution**: Team members often spend hours troubleshooting an issue, only to realize that the same problem was already solved a week earlier by another colleague.
+- **Time-Consuming Repetitive Tickets**: Around 60% of your time is wasted on repetitive tickets because end users struggle to perform basic operations on their own.
+
+### The Promise of LLMs: A Beacon of Hope
+
+The advent of Large Language Models (LLMs) offers a game-changing solution. You envision building a chatbot that leverages Retrieval-Augmented Generation (RAG) to search through previously resolved tickets. This would help both your team members and end users:
+
+- **Team Members** could quickly find solutions without duplicating efforts.
+- **End Users** could resolve their issues independently, reducing ticket volume.
+
+### The Nightmare of PII Exposure
+
+Tickets in your system contain a wide range of Personal Identifiable Information (PII): names, roles, passwords, and other sensitive data. How can you ensure that sensitive information is never retrieved, avoiding catastrophic data leaks within your company?
+
+### Enter Eunomia
+
+Eunomia’s advanced PII management instruments ensure that sensitive data—such as names, emails, and phone numbers—are effectively safeguarded.
+
+#### Processing Methods
+
+Eunomia can operate at various stages within data pipelines, offering seamless integration with diverse architectures based on the developer's preferred integration point. It can be integrated in both:
+
+1. **In One Go Processing**: Sensitive data is identified and removed in advance, storing a static "sanitized" version of the ticket directly in the database.
+2. **On-Demand Processing**: When a ticket is retrieved by the RAG system, Eunomia dynamically cleans the ticket before delivering the response.
+
+This dual approach ensures that PII is never exposed, allowing you to maintain robust data governance while harnessing the full power of LLM-based systems.
+
+#### Example: Original Ticket vs. Processed Ticket
+
+**Original Ticket:**
+```plaintext
+Ticket ID: #IT-2024-0001
+User Information:
+Name: John Doe
+Email: john.doe@example.com
+Phone Number: +1 (555) 123-4567
+```
+
+**Processed Ticket:**
+```plaintext
+Ticket ID: #IT-2024-0001
+User Information:
+Name: <PERSON>
+Email: <EMAIL_ADDRESS>
+Phone Number: <PHONE_NUMBER>
+```
+
+### Developing the Solution with Eunomia
+
+#### Step 1: Setup
+
+To begin, ensure that Eunomia is installed by following the [installation guide](../get_started/installation.md). Then, import the necessary components in your script:
+
+```python
+from eunomia.orchestra import Orchestra
+from eunomia.instruments import PiiInstrument
+```
+You can find out more about the components used at:
+
+- [Orchestra docs](../api/orchestra.md)
+- [PiiInstrument docs](../api/instruments/pii_instrument.md)
```

Comment:
Bullet points take a lot of space here, I would rather list them in the same paragraph